### PR TITLE
Fix NullPointerException in YouTubeStreamExtractor.getStreamSegments()

### PR DESCRIPTION
**Pull request by TobiGr**
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I have tested the API against [NewPipe](https://github.com/TeamNewPipe/NewPipe).
- [x] I agree to create a pull request for [NewPipe](https://github.com/TeamNewPipe/NewPipe) as soon as possible to make it compatible with the changed API.

Fixes https://github.com/TeamNewPipe/NewPipe/issues/7792

For more info see https://github.com/TeamNewPipe/NewPipe/issues/7792#issuecomment-1030900188

APK: [app-debug.zip](https://github.com/TeamNewPipe/NewPipeExtractor/files/8049631/app-debug.zip)

### CODE REVIEW

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java [hunk 1]
```diff
@@ -1272,7 +1272,7 @@ public Locale getLanguageInfo() {
                 final String panelIdentifier = panels.getObject(i)
                         .getObject("engagementPanelSectionListRenderer")
                         .getString("panelIdentifier");
-                if (panelIdentifier.equals(
+                if (panelIdentifier != null && panelIdentifier.equals(
                         "engagement-panel-macro-markers-description-chapters")) {
```
Comment (litetex) [inline extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java:1276]:
```suggestion
                if ("engagement-panel-macro-markers-description-chapters".equals(panelIdentifier)) {
```

**Review by litetex — COMMENTED** [review]
This would fix the problem with less code

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java (hunk 1 above)
Comment (TobiGr) [reply extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java]: That's way better.

**Review by litetex — APPROVED** [review]

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java [hunk 2]
```diff
@@ -1272,8 +1272,7 @@ public Locale getLanguageInfo() {
                 final String panelIdentifier = panels.getObject(i)
                         .getObject("engagementPanelSectionListRenderer")
                         .getString("panelIdentifier");
-                if (panelIdentifier.equals(
-                        "engagement-panel-macro-markers-description-chapters")) {
+                if ("engagement-panel-macro-markers-description-chapters".equals(panelIdentifier)) {
```
Comment (mhmdanas) [inline extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java:1275]: It's not obvious at all why the comparison is flipped if you don't know `panelIdentifier` can be `null`. How about adding a comment?

Comment (TobiGr) [reply extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java:1275]: Something like that?
``` java
                // panelIdentifier might be null if the panel has something to do with ads
                // See https://github.com/TeamNewPipe/NewPipe/issues/7792#issuecomment-1030900188
                @Nullable
                if ("engagement-panel-macro-markers-description-chapters".equals(panelIdentifier)) {
```

Comment (TobiGr) [reply extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java:1275]: because that var can be null

Comment (mhmdanas) [reply extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java:1275]: Ah, I saw the comment before you edited it; the `@Nullable` was directly before the `if` statement then. xD

**Pull request merged**